nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -467,7 +467,7 @@ in
         fi
       '';
 
-    nix.nrBuildUsers = mkDefault (lib.max 32 cfg.maxJobs);
+    nix.nrBuildUsers = mkDefault (lib.max 32 (if cfg.maxJobs == "auto" then 0 else cfg.maxJobs));
```

Comment:
This option becomes interesting once you have more then 32 hyper threads/cores :)